modules/webgl/src/features/features.js

```diff
@@ -37,14 +36,36 @@ export function getFeatures(gl) {
 
 // TODO - cache the value
 function isFeatureSupported(gl, cap) {
+  gl.luma = gl.luma || {};
+  gl.luma.caps = gl.luma.caps || {};
+
+  let isSupported;
+
+  if (gl.luma.caps[cap] !== undefined) {
```

Comment:
@tsherif , I was thinking, for feature overrides, we can add `gl.luma.overrideCaps` and check that here first and if set use it.  We can have `setFeatureOverrides(caps)` and `clearFeatureOverrides()` methods to update `gl.luma.overrideCaps`

Reply:
Yeah this is a nice idea. I could imagine an API like this:

```js

withFeatureOverrides(caps, () => {
    // Do your thing
});
```
This could use the cache to temporarily override feature checks. (I'd worry about allowing feature overrides to exist long term as it could lead to super hard-to-track-down bugs).

Reply:
provided we are querying and drawing within that block. We might still need individual methods to set/re-set overrides for uses cases such as deck.gl Layers, where querying happens during layer init time and drawing happens later.

Reply:
I don't think this is something that should be used except for unit testing.